core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -86,7 +86,10 @@ private[spark] object TaskDescription {
     dataOut.writeInt(taskDescription.properties.size())
     taskDescription.properties.asScala.foreach { case (key, value) =>
       dataOut.writeUTF(key)
-      dataOut.writeUTF(value)
+      // SPARK-19796 -- writeUTF doesn't work for long strings, which can happen for property values
+      val bytes = value.getBytes("utf-8")
+      dataOut.writeInt(bytes.length)
+      dataOut.write(bytes)
```

Comment:
(here's the code: http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/io/ObjectOutputStream.java#1160. I guess one downside of that is that it's more time-consuming because it does some extra steps, but I'm not sure that is a huge issue)